sdk/keyvault/keyvault-certificates/test/public/identifier.spec.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as assert from "assert";
+import { KeyVaultCertificatesIdentifier } from "../../src";
+
+describe("KeyVault Certificates Identifier", () => {
+  it("It should work with a URI of a certificate before it gets a version", async function() {
+    const uri = "https://keyvault-name.vault.azure.net/certificates/certificate-name/pending";
+    const identifier = new KeyVaultCertificatesIdentifier(uri);
+
+    assert.deepEqual(identifier, {
+      id: "https://keyvault-name.vault.azure.net/certificates/certificate-name/pending",
+      vaultUrl: "https://keyvault-name.vault.azure.net",
+      collection: "certificates",
+      name: "certificate-name",
+      version: "pending"
+    });
+  });
+
+  it("It should work with a URI of a certificate with a specific version", async function() {
+    const uri = "https://keyvault-name.vault.azure.net/certificates/certificate-name/version";
+    const identifier = new KeyVaultCertificatesIdentifier(uri);
+
+    assert.deepEqual(identifier, {
+      id: "https://keyvault-name.vault.azure.net/certificates/certificate-name/version",
+      vaultUrl: "https://keyvault-name.vault.azure.net",
+      collection: "certificates",
+      name: "certificate-name",
+      version: "version"
+    });
+  });
+
+  it("It should work with the URI of a certificate's issuer", async function() {
```

Comment:
It really shouldn't. The name and version are not names and versions. As discussed offline, let's only support (deleted) certificates, keys, and secrets right now.

Reply:
If you want to use it that way internally, fine (maybe not that maintainable, but I understand), but let's not expose that publicly.

Reply:
I'll clean this up! thanks!